File: Carrito-de-Compras/02-Fase2-POO-BlueJ/Proyecto-BlueJ/Factura.java
public class Factura
{
    private int numeroFactura;
    private Pedido pedido;
    private String fecha;
    private double total;

    public Factura(int numeroFactura, Pedido pedido, String fecha)
    {
        this.numeroFactura = numeroFactura;
        this.pedido = pedido;
        this.fecha = fecha;
        this.total = pedido.getTotal();
    }

    public void generarFactura()
    {
        System.out.println("Factura generada correctamente.");
    }

    public void mostrarFactura()
    {
        System.out.println("=== FACTURA ===");
        System.out.println("Numero: " + numeroFactura);
        System.out.println("Fecha: " + fecha);
        System.out.println("Cliente: " + pedido.getCliente().getNombre());
        System.out.println("Pedido: " + pedido.getIdPedido());
        System.out.println("Total: Q" + total);
    }
}

File: Carrito-de-Compras/03-Fase3-CSharp/Models/Factura.cs
namespace CarritoCompras.Models;

public class Factura
{
    public int NumeroFactura { get; private set; }
    public Pedido Pedido { get; private set; }
    public DateTime Fecha { get; private set; }

    public Factura(int numeroFactura, Pedido pedido)
    {
        NumeroFactura = numeroFactura;
        Pedido = pedido;
        Fecha = DateTime.Now;
    }

    public void Mostrar()
    {
        Console.WriteLine("\n========== FACTURA ==========");
        Console.WriteLine($"Factura No.: {NumeroFactura}");
        Console.WriteLine($"Fecha: {Fecha:dd/MM/yyyy HH:mm}");
        Console.WriteLine($"Cliente: {Pedido.Cliente.Nombre}");
        Console.WriteLine($"Pedido No.: {Pedido.IdPedido}");

        foreach (var detalle in Pedido.Detalles)
        {
            Console.WriteLine(
                $"{detalle.Producto.Nombre} x {detalle.Cantidad} = Q{detalle.Subtotal:0.00}");
        }

        Console.WriteLine($"TOTAL: Q{Pedido.Total:0.00}");
        Console.WriteLine("=============================\n");
    }
}
